.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,60 @@
+name: To Do app CI/CD
+on: [push]
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    env:
+        TOKEN: ${{ secrets.TOKEN }}
+        SECRET_KEY: ${{ secrets.SECRET_KEY }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build
+      run: docker build --target test --tag todo-app:test .
+
+    - name: Unit/Integration Test
+      run: docker run todo-app:test tests
+    
+    - name: End to End Test
+      run: docker run -e SECRET_KEY -e TOKEN todo-app:test tests_e2e
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    env:
+        docker_hub_password: ${{ secrets.docker_hub_password }}
+        docker_hub_username: ${{ secrets.docker_hub_username }}
+        HEROKU_API_KEY: ${{secrets.HEROKU_API_KEY}}
+    needs: build
+    # if: github.ref == 'refs/heads/master' # include this line if you want to limit the deployment to the master branch
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Login to DockerHub
+      run: echo $docker_hub_password | docker login --username $docker_hub_username --password-stdin
+    
+    - name: Build the Docker image
+      run: docker build --target production --tag sjain309/todo-app-prod:latest .
+
+    - name: Push the Docker image to DockerHub
+      run: docker push sjain309/todo-app-prod:latest
+
+    - name: Pull the Docker image from DockerHub
+      run: docker pull sjain309/todo-app-prod
```

Comment:
Is there any point pulling this right after having pushed it? We could skip this step and we should see exactly the same behaviour (barring very unlucky race conditions - we shouldn't be  pushing to the latest DockerHub image multiple times a second!)